query/src/lib.rs

```diff
@@ -486,18 +486,61 @@ impl PatternValuePlace {
     }
 }
 
-/*
-pub enum PullPattern {
-    Constant(Constant),
-    Variable(Variable),
+// Not yet used.
+// pub enum PullDefaultValue {
+//     EntidOrInteger(i64),
+//     IdentOrKeyword(Rc<NamespacedKeyword>),
+//     Constant(NonIntegerConstant),
+// }
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum PullConcreteAttribute {
+    Ident(Rc<NamespacedKeyword>),
+    Entid(i64),
+}
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum PullAttributeSpec {
```

Comment:
I think your review comment boils down to: `Pull` should be

```rust
pub enum Pull {
    Wildcard,
    Attributes(Vec<PullAttributeSpec>),
}
```

and we should lose the `Wildcard` option here, right?

Datomic allows composition of wildcarding and specified attributes:

```clojure
(pull ?x [*, {:foo/friend [:foo/name]}])
```

… which means "grab everything, and additionally traverse `:foo/friend` to find names".

It also doesn't forbid inclusion of the wildcard more than once.

The structures in `mentat_query` represent user inputs. Pull requires pre-processing before it can be sensibly executed.